packages/api/controllers/application.js

```diff
@@ -276,3 +293,19 @@ function validateProperties(request, next) {
   }
   return true;
 }
+
+
+function getIdentifier(req) {
+  return encodeURIComponent(req)
+    .toLowerCase()
+    /* Replace the encoded hexadecimal code with `-` */
+    .replace(/%[0-9a-zA-Z]{2}/g, '-')
+    /* Replace any special characters with `-` */
+    .replace(/[\ \-\/\:\@\[\]\`\{\~\.]+/g, '-')
+    /* Special characters are replaced by an underscore */
+    .replace(/[\|!@#$%^&*;"<>\(\)\+,]/g, '_')
+    /* Remove any starting or ending `-` */
+    .replace(/^-+|-+$/g, '')
+    /* Removing multiple consecutive `-`s */
+    .replace(/--+/g, '-');
+}
```

Comment:
@soumyadip007 do test this function before using it

Reply:
Yes @hybridx . I've tested it.